### ISSUE TITLE
It is not returning caption

### ISSUE THREAD

**Issue by yucelce** [devhims/youtube-caption-extractor#8]
It is not returning caption for any video, it returns only this message : "Enjoy the videos and music you love, upload original content, and share it all with friends, family, and the world on YouTube."

**Comment by devhims** [comment]
@yucelce Unfortunately, YouTube has recently implemented anti-scraping policies that mandate sign in to access video data in backend environments.

**Comment by cherfaoui-syphax** [comment]
> @yucelce Unfortunately, YouTube has recently implemented anti-scraping policies that mandate sign in to access video data in backend environments.

I can confirm, i tried all the major youtube caption scraping libraries in Python, Go, and Js and all of them stopped working. even when i used residential proxies. This library still works fine on client side. in a chrome extension. this is probably the only use of this library because of CORS issues.

**Comment by xpluscal** [comment]
@devhims whats your stance on use then?
Aka is there an alternative?

Can you run this on the client and bypass CORS?

**Comment by cherfaoui-syphax** [comment]
> [@devhims](https://github.com/devhims) whats your stance on use then? Aka is there an alternative?
> 
> Can you run this on the client and bypass CORS?

Yeah it works if you run it client side inside the youtube window, i'm using it currently inside a content script of a chrome extension and it works fine.

**Comment by pablo-s-dev** [comment]
> [@yucelce](https://github.com/yucelce) Unfortunately, YouTube has recently implemented anti-scraping policies that mandate sign in to access video data in backend environments.

How can websites like savefromnet still do it? Residential servers? From an ec2 instance is not possible right? even logged in?

**Comment by cherfaoui-syphax** [comment]
> > [@yucelce](https://github.com/yucelce) Unfortunately, YouTube has recently implemented anti-scraping policies that mandate sign in to access video data in backend environments.
> 
> How can websites like savefromnet still do it? Residential servers? From an ec2 instance is not possible right? even logged in?
Youtube is probably relying on something like cookies to make sure you're not using a bot so it should be possible to still get the captions if you simulate a browser session with something like puppeteer , or selenium or cloudflare browser rendering. i haven't actually tried it since my use case doesn't call for it but since there are still websites where you can get the captions ther's a way and you can probably do it if you dig deep enough

**Comment by devhims** [comment]
v1.8.2 fixes the issue and should bypass YouTube bot blocking.
